src/Mailgun/Api/Suppressions/Bounce.php

```diff
@@ -0,0 +1,155 @@
+use Mailgun\Resource\Api\Suppressions\Bounce\DeleteResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\IndexResponse;
+use Mailgun\Resource\Api\Suppressions\Bounce\ShowResponse;
+
+/**
+ * @author Sean Johnson <sean@mailgun.com>
+ */
+class Bounce extends HttpApi
+{
+    use Pagination;
+
+    /**
+     * @param string $domain
+     *
+     * @return IndexResponse
+     */
+    public function index($domain, $limit = 100)
+    {
+        Assert::notEmpty($domain);
+        Assert::range($limit, 1, 10000, 'Limit parameter must be between 1 and 10000');
+
+        $response = $this->httpGet(sprintf('/v3/%s/bounces', $domain));
+
+        return $this->safeDeserialize($response, IndexResponse::class);
+    }
+
+    /**
+     * @param string $domain
+     * @param string $address
+     *
+     * @return ShowResponse
+     */
+    public function show($domain, $address)
+    {
+        Assert::notEmpty($domain);
+        Assert::notEmpty($address);
+        $response = $this->httpGet(sprintf('/v3/%s/bounces/%s', $domain, $address));
+
+        return $this->safeDeserialize($response, ShowResponse::class);
+    }
+
+    /**
+     * @param string    $domain
+     * @param string    $address
+     * @param string    $code      optional
+     * @param string    $error     optional
+     * @param \DateTime $createdAt optional
+     *
+     * @return CreateResponse
+     */
+    public function create($domain, $address, $code = null, $error = null, $createdAt = null)
+    {
+        Assert::notEmpty($domain);
+        Assert::notEmpty($address);
+
+        $params = [
+            'address' => $address,
+        ];
+
+        foreach (['code' => $code, 'error' => $error, 'created_at' => $createdAt] as $k => $v) {
+            if (!empty($v)) {
+                $params[$k] = $v;
+            }
+        }
+
+        $response = $this->httpPost(sprintf('/v3/%s/bounces', $domain), $params);
+
+        return $this->safeDeserialize($response, CreateResponse::class);
+    }
+
+    /**
+     * @param string $domain
+     * @param string $address
+     *
+     * @return DeleteResponse
+     */
+    public function delete($domain, $address)
+    {
+        Assert::notEmpty($domain);
+        Assert::notEmpty($address);
+        $response = $this->httpDelete(sprintf('/v3/%s/bounces/%s', $domain, $address));
+
+        return $this->safeDeserialize($response, DeleteResponse::class);
+    }
+
+    /**
+     * @param string $domain
+     *
+     * @return DeleteResponse
+     */
+    public function deleteAll($domain)
+    {
+        Assert::notEmpty($domain);
+        $response = $this->httpDelete(sprintf('/v3/%s/bounces', $domain));
+
+        return $this->safeDeserialize($response, DeleteResponse::class);
+    }
+
+    /*
+     * INDEX PAGINATION
+     */
+
+    /**
+     * @param IndexResponse $index
+     *
+     * @return IndexResponse|null
+     */
+    public function getPaginationNext(IndexResponse $index)
```

Comment:
After a quick look - and maybe I am wrong - looks like these methods are duplicated: we're calling 12 times on this PR to `$this->getPaginationUrl(...)` to call another 3 times each of these calls: `getNextUrl()`, `getPreviousUrl()`, `getFirstUrl()` & `getLastUrl()`. I am sure there is a better way to do this and to avoid duplicated code.

Question: cannot this be moved to the `Pagination` object instead? Maybe we can set-up these these 4 methods and then ask just for the `IndexResponse::class` that we want to use as an argument?